akka-diagnostics/src/main/scala/akka/diagnostics/StarvationDetector.scala

```diff
@@ -144,19 +148,46 @@ object StarvationDetector {
       config,
       () => system.whenTerminated.isCompleted)
 
+  /**
+   * Creates and runs a StarvationDetector thread for the internal dispatcher of the system. Akka 2.6 introduced this
+   * dispatcher for internal tasks like clustering. If there is no internal dispatcher, does nothing.
+   */
+  def checkInternalDispatcher(system: ActorSystem): Unit =
+    checkInternalDispatcher(
+      system,
+      StarvationDetectorSettings.fromConfig(system.settings.config.getConfig("akka.diagnostics.starvation-detector")))
+
+  /**
+   * Creates and runs a StarvationDetector thread for the internal dispatcher of the system with custom configuration.
+   * Akka 2.6 introduced this dispatcher for internal tasks like clustering. If there is no internal dispatcher, does
+   * nothing.
+   */
+  def checkInternalDispatcher(system: ActorSystem, config: StarvationDetectorSettings): Unit = {
+    val internalDispatcherPath = "akka.actor.internal-dispatcher"
+    val dispatcher = system.dispatchers.lookup(internalDispatcherPath)
+    checkExecutionContext(
+      dispatcher,
+      Logging(system, classOf[StarvationDetectorThread]),
+      config,
+      () => system.whenTerminated.isCompleted)
+  }
+
   /**
    * Creates and runs a StarvationDetector thread for the given ExecutionContext. Thread analytics are currently only
    * available for Akka dispatchers.
    *
    * You need to provide a `hasTerminated` function that will be used to figure out if the execution context has shut
    * down to shutdown the starvation detector thread.
+   *
+   * Note that an ExecutionContext will only have one StarvationDetector for it active at a time. If there is another
+   * StarvationDetector running, this method does nothing.
```

Comment:
Having at times gone crazy adding StarvationDetectors to already-monitored contexts, this is welcome!